src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatelessMarshallingStrategy.cs

```diff
@@ -82,11 +82,19 @@ public IEnumerable<StatementSyntax> GenerateMarshalStatements(TypePositionInfo i
             }
 
             // <nativeIdentifier> = <convertToUnmanaged>;
-            yield return ExpressionStatement(
-                AssignmentExpression(
+            var assignment = AssignmentExpression(
                     SyntaxKind.SimpleAssignmentExpression,
                     IdentifierName(nativeIdentifier),
-                    convertToUnmanaged));
+                    convertToUnmanaged);
+
+
+            if (_unmanagedType is PointerTypeInfo pointer)
+            {
+                var rewriter = new NonBlittableElementsMarshalling.PointerNativeTypeAssignmentRewriter(assignment.Right.ToString(), (PointerTypeSyntax)pointer.Syntax);
```

Comment:
I think this was in a separate PR, but move `PointerNativeTypeAssignmentRewriter` to a non-nested type?